### PR TITLE
[GPU] Fixes for GPUs from other vendors

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - Fixed several OCL kernels which had unconventional syntax
 - Move flops calculation to device class and added check for vendor id there

### CODE REVIEW

**Comment by akladiev** [comment]
The valid link to the validation results: https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/github_trigger/job/openvino/job/PR-12577/3/

File: src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp [hunk 1]
```diff
@@ -91,47 +92,58 @@ static std::vector<cl::Device> getSubDevices(cl::Device& rootDevice) {
     return subDevices;
 }
 
+std::vector<device::ptr> ocl_device_detector::sort_devices(const std::vector<device::ptr>& devices_list) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp:95]: Why dont we just add some weight for integrated_gpu/INTEL_GPU/others and then sort it using some priority queue or etc?

File: src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp [hunk 2]
```diff
@@ -91,47 +92,58 @@ static std::vector<cl::Device> getSubDevices(cl::Device& rootDevice) {
...
+    std::vector<device::ptr> sorted_list;
+    size_t insert_idx = 0;
+    for (auto& dptr : devices_list) {
+        if (dptr->get_info().dev_type == cldnn::device_type::integrated_gpu) {
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp:99]: Very theoretical situation, but anyway.. Shouldn't we make sure that it is Intel device as well before moving it at first position?

**Review by sshlyapn — APPROVED** [review]

File: src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp (hunk 2 above)
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp]: changed

File: src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp (hunk 1 above)
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp:110]: updated

Comment (yeonbok) [reply src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp:110]: If the priority is same, (e.g., multiple INTEL dgpu devices ?) should it be sorted as inserted order?

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp:110]: I think it's not really critical, but changed to stable_sort

**Pull request merged**